models/dalec/R/met2model.DALEC.R

```diff
@@ -58,109 +58,154 @@ met2model.DALEC <- function(in.path, in.prefix, outfolder, start_date, end_date,
-    LeafWaterPot = rep(LeafWaterPot,length(Tair))
+  ## check to see if the outfolder is defined, if not create directory for output
+  if(!file.exists(outfolder)){
+    dir.create(outfolder)
   }
+  
+  out <- NULL
+  
+  # get start/end year since inputs are specified on year basis
+  start_year <- year(start_date)
+  end_year <- year(end_date)
+  
+  ## loop over files
+  # TODO need to filter out the data that is not inside start_date, end_date
+  for(year in start_year:end_year) {
+    old.file <- file.path(in.path, paste(in.prefix, year, "nc", sep="."))
     
-  ##build day of year
-  doy <- rep(1:365,each=86400/dt)
-  if(year %% 4 == 0){  ## is leap
-      doy <- rep(1:366,each=86400/dt)
-  }
-
-  ## Aggregate variables up to daily
-  Tmean = udunits2::ud.convert(tapply(Tair,doy,mean,na.rm=TRUE),"Kelvin","Celsius")
-  Tmin  = udunits2::ud.convert(tapply(Tair,doy,min,na.rm=TRUE),"Kelvin","Celsius")
-  Tmax  = udunits2::ud.convert(tapply(Tair,doy,max,na.rm=TRUE),"Kelvin","Celsius")
-  Rin   = tapply(SW,doy,sum)*dt*1e-6 # J/m2/s * s * MJ/J
-  LeafWaterPot = tapply(LeafWaterPot,doy,mean)
-  CO2   = tapply(CO2,doy,mean)
-  HydResist = tapply(HydResist,doy,mean)
-  leafN = tapply(leafN,doy,mean)
-  doy   = tapply(doy,doy,mean)
+    ## open netcdf
+    nc <- nc_open(old.file)
+    
+    ## convert time to seconds
+    sec   <- nc$dim$time$vals  
+    sec = udunits2::ud.convert(sec,unlist(strsplit(nc$dim$time$units," "))[1],"seconds")
+    
+    ifelse(leap_year(year)==TRUE,
+           dt <- (366*24*60*60)/length(sec), #leap year
+           dt <- (365*24*60*60)/length(sec)) #non-leap year
+    tstep = 86400/dt
+    
+    ## extract variables
+    lat  <- ncvar_get(nc,"latitude")
+    lon  <- ncvar_get(nc,"longitude")
+    Tair <- ncvar_get(nc,"air_temperature")  ## in Kelvin
+    SW   <- ncvar_get(nc,"surface_downwelling_shortwave_flux_in_air") ## in W/m2
+    CO2  <- try(ncvar_get(nc,"mole_fraction_of_carbon_dioxide_in_air"))
+    nc_close(nc)
+    
+    useCO2 = is.numeric(CO2)  
+    if(useCO2)  CO2 <- CO2/1e6  ## convert from mole fraction (kg/kg) to ppm
+    
+    
+    ## is CO2 present?
+    if(!is.numeric(CO2)){
+      logger.warn("CO2 not found in",old.file,"setting to default: 400 ppm")
+      CO2 = rep(400,length(Tair))
+    }
+    
+    ## check to see if the outfolder is defined, if not create directory for output
+    if(!file.exists(outfolder)){
+      dir.create(outfolder)
+    }
+    
+    out <- NULL
+    
+    # get start/end year since inputs are specified on year basis
+    start_year <- year(start_date)
+    end_year <- year(end_date)
+    
+    ## loop over files
+    # TODO need to filter out the data that is not inside start_date, end_date
+    for(year in start_year:end_year) {
+      old.file <- file.path(in.path, paste(in.prefix, year, "nc", sep="."))
+      
+      if(!file.exists(old.file)){logger.warn(msg = paste(old.file,"does not exist"))
```

Comment:
one option is to you use the next keyword here, so you don't need the else. next will jump to the next iteration of the for loop, break will break out of the for loop.